statistics/histogram.go

```diff
@@ -1665,9 +1665,9 @@ func mergeBucketNDV(sc *stmtctx.StatementContext, left *bucket4Merging, right *b
 //		repeat = sum of buckets[i] (buckets[i].upper == global bucket.upper && i in [l...r))
 //		ndv = merge bucket ndv from r-1 to l by mergeBucketNDV
 // Notice: lower is not calculated here.
-func mergePartitionBuckets(sc *stmtctx.StatementContext, buckets []*bucket4Merging) *bucket4Merging {
+func mergePartitionBuckets(sc *stmtctx.StatementContext, buckets []*bucket4Merging) (*bucket4Merging, error) {
 	if len(buckets) == 0 {
-		return nil
+		return nil, errors.Errorf("not enough buckets to merge")
```

Comment:
Should we return the error here? Or check whether `bucket4Merging` is nil in the calling function. What's your opinion? @qw4990 @rebelice @xuyifangreeneyes